### PR TITLE
Countries without post-codes now working

### LINKED ISSUE #1518: Some countries without Post-Code still have Post-Code set as required.
**Describe the bug**
Based on feedback from a user, they have attempted to make a donation with Hong Kong set as the country, which removes the post code field from view and should make the field not required so the user can submit their donation.

However, if you fill out the form and attempt to make a donation, you will recieve an error message, and when changing to country with post code, will see that it is still expecting a post code. 

**To Reproduce**
1. Go to the donation homepage
2. Fill out the form
3. Submit the form

**Expected behavior**
It should allow the user to submit their donation without a post code

### PR DESCRIPTION
**Pull request by danielfmiranda**
Closes #1518 
Related PRs/issues: #1535  (Investigatory PR)

**Steps to test**: 

1. Visit https://donate-wagta-postal-cod-ahiric.herokuapp.com/en-US/
2. Make a selection from one of the donation amounts, or enter a custom value.
3. Click the purple credit card payment button
4. Fill out the form, using the test credit card info below, and selecting "Hong Kong" as the country.
5. Click "Donate" and you should get redirected to the thank you page.
6. If redirected to thank you page, testing is done! 

**TEST CC INFO**:
Card Number: 4111 1111 1111 1111
EXP: 11/23
CVV: 123

## Checklist

**Changes in Models:**
- [x] [Are my changes backward-compatible](https://github.com/mozilla/foundation.mozilla.org/blob/master/docs/workflow.md#django-migrations-what-to-do-when-working-on-backward-incompatible-migrations). If not, did I schedule a deploy with the rest of the team?

### CODE REVIEW

File: donate/payments/forms.py [hunk 1]
```diff
@@ -13,14 +13,18 @@
...
+# Loading a JSON list of countries and their respective post code formats if applicable,
+# from the source/js directory.
 try:
-    with open('./source/js/components/post-codes-list.json') as post_code_data:
+    with open(f'{root}/source/js/components/post-codes-list.json') as post_code_data:
         COUNTRY_POST_CODES = json.load(post_code_data)
-except Exception:
+except Exception as error:
+    print('ERROR: could not read in post codes list')
+    print(error)
```
Comment (Pomax) [inline donate/payments/forms.py:27]: Let's switch this to using the `logger` like in https://github.com/mozilla/donate-wagtail/blob/master/donate/payments/tasks.py ?

File: donate/payments/forms.py [hunk 2]
```diff
@@ -72,6 +76,7 @@ def clean(self):
         country = cleaned_data.get('country', '')
         # If we cannot import post-code data, default to it being required.
         if COUNTRY_POST_CODES is None:
+            print('Error: no postal code data available, defaulting to required(postal_code).')
```
Comment (Pomax) [inline donate/payments/forms.py:79]: probably here, too

**Review by Pomax — CHANGES_REQUESTED** [review]

File: donate/payments/forms.py (hunk 1 above)
Comment (danielfmiranda) [reply donate/payments/forms.py]: Done! Marking as resolved :D

**Review by Pomax — APPROVED** [review]
yay!

**Pull request merged**